### ISSUE TITLE
Reconsider #44 (change user and group)

### ISSUE THREAD

**Issue by blackgnezdo** [haskell/process#230]
The capability to change user and group was added 6 years ago. There a few reasons why I believe this feature may not be pulling its weight:

* doing setuid dance *correctly* in a cross-platform way is hard (consider how [much effort openssh-portable goes through](https://github.com/openssh/openssh-portable/blob/master/openbsd-compat/bsd-setres_id.c))
* there's a TODO acknowledging the questionable [correctness of the implementation](https://github.com/haskell/process/blob/master/cbits/posix/fork_exec.c#L175-L176)
* posix_spawn API doesn't support this capability so this feature requires keeping both code paths and runtime fall-back which is unlikely to be a well-tested scenario.
* no tests cover this functionality (because it is really inconvenient to test)

@jprider63 do you still use this feature that you added? Can somebody can think of a way to survey the ecosystem to figure out how much usage the feature received?

In summary, I believe the feature is likely not secure, contains race conditions, complicates the codebase, and has no tests. This seems like a good candidate for removal.

**Comment by jprider63** [comment]
Hi @blackgnezdo. Yes, `child_user` and `child_group` are still used in internal code.

**Comment by bgamari** [comment]
For what it's worth, I tend to agree that user and group changing isn't pulling its weight. This sort of functionality is extremely platform-dependent and ripe for nasty (potentially exploitable) bugs (e.g. #149).
